### PR TITLE
Added Neon_number java file and updated the READMe.md file 

### LINKED ISSUE #2463: Neon number  under mathematics section
## 💥 Proposal

A neon number is a number where the sum of digits of square of the number is equal to the number. The task is to check and print neon numbers in a range.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by ananddasani**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes i have read the Contributing Guidelines on Pull Requests

### Description

I have solved the issue #2463 (Neon number using java), which was assigned to me by @siddharth25pandey 

So i request project admins @siddharth25pandey  , @bhav09 to please review my PR.

Fixes #2463 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

I would like to request @HarshCasper , @siddharth25pandey , @bhav09  to please assign me 

issue #2223 Cycle sort 

as it is not assigned to anyone till now and i have requested, i hope you hear me :)

### CODE REVIEW

**Comment by bhav09** [comment]
Great work! Keep it up :)

File: Java/math/Neon_number.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Program by :: Anand Dasani
+Date :: 13-03-2021
+
+Aim :: To check the given number is Neon or not.
+
+What is Neon number ?
+A Neon number is a number where the sum of digits of square of the number is equal to that number.
+
+                        ALGO
+1. Take a number form the user
```
Comment (gat786) [inline Java/math/Neon_number.java:11]: From* and not form

**Review by bhav09 — COMMENTED** [review]
A small amendment:
add a new file at the end of the file. Rest, well to go :)

**Comment by ananddasani** [comment]
> A small amendment:
> add a new file at the end of the file. Rest, well to go :)

@bhav09 sir , will you please tell me in details .. so that i didn't found this type of error anymore from next time onwards :)

I am also getting DeepSource: Java (beta) error , 
will you please explain me , the solution and precaution steps ......

File: Java/math/Neon_number.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Enter the number :: 1   // input
+9 is Neon number :)     // output
+
+Test case :: T = 4
+Enter the number :: 3   // input
+9 is not a Neon number :(     // output
+
+Time complexity :: O(log(n))
+Space complexity :: O(1)
+ */
```
Comment (HarshCasper) [inline Java/math/Neon_number.java:86]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Pull request closed without merging**